packages/@uppy/tus/src/index.js

```diff
@@ -252,6 +259,24 @@ module.exports = class Tus extends BasePlugin {
         resolve(upload)
       }
 
+      uploadOptions.onShouldRetry = (err, retryAttempt, options) => {
+        if (err?.originalResponse?.getStatus() === 429) {
+          if (!this.requests.isPaused) {
+            const { done, value } = this.#retryDelayIterator.next()
+            if (done) {
+              return false
+            }
+            this.requests.pause(value)
+          }
+          queuedRequest = this.requests.run(queueRequest)
+        } else {
+          setTimeout(() => {
+            queuedRequest = this.requests.run(queuedRequest)
+          }, options.retryDelays[retryAttempt])
```

Comment:
I'd expect `retryAttempt` to start with 1, as in the "first" attempt to retry. Also means if retryAttempt is 4, there were 4 retry attempts. Just reads more easily.

But `options.retryDelays` is a 0-indexed array. Might be useful to add a comment here that the first retry attempt means `retryAttempt === 0` (if that is the case, ofc).

Reply:
`retryAttempt` starts at `0`, because before on the first attempt, it's not a retry.
Looking at the https://github.com/tus/tus-js-client/blob/8574294c30765dbe3a71a7a066a88b060e0e84d6/lib/upload.js#L453, it looks like if we return `false` the retryAttempt will always be `0`, so we need to hack our way around it..